From 7c165ead8fe6a7adaa98c409242db1ec3386fa9f Mon Sep 17 00:00:00 2001
From: Pranav Vashi <neobuddy89@gmail.com>
Date: Tue, 11 Aug 2026 09:40:12 +0000
Subject: [PATCH] SystemUI: Use themewrapper for QSCustomizer and tune
 colorUnavailable

* Restores behvavior from A13 for colorUnavailable, shadeDisabled is now unused.

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>

[Port to A16] Same edits; A16 added blur-handling lines between anchors.

Change-Id: Ie6fca3e68c090a8410c19b3f9429bb051a3b5bce
---
 .../com/android/systemui/qs/customize/QSCustomizer.java  | 9 ++++++---
 .../com/android/systemui/qs/tileimpl/QSTileViewImpl.kt   | 2 +-
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/customize/QSCustomizer.java b/packages/SystemUI/src/com/android/systemui/qs/customize/QSCustomizer.java
index 468d687cb81d..3904590d2662 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/customize/QSCustomizer.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/customize/QSCustomizer.java
@@ -25,6 +25,7 @@ import android.content.res.Configuration;
 import android.graphics.Color;
 import android.util.AttributeSet;
 import android.util.TypedValue;
+import android.view.ContextThemeWrapper;
 import android.view.LayoutInflater;
 import android.view.Menu;
 import android.view.MenuItem;
@@ -73,7 +74,9 @@ public class QSCustomizer extends LinearLayout {
     public QSCustomizer(Context context, AttributeSet attrs) {
         super(context, attrs);
 
-        LayoutInflater.from(getContext()).inflate(R.layout.qs_customize_panel_content, this);
+        Context themedContext =
+                new ContextThemeWrapper(context, R.style.Theme_SystemUI_QuickSettings);
+        LayoutInflater.from(themedContext).inflate(R.layout.qs_customize_panel_content, this);
         View customizeContainer = findViewById(R.id.customize_container);
         if (notificationShadeBlur()) {
             customizeContainer
@@ -85,9 +88,9 @@ public class QSCustomizer extends LinearLayout {
             mToolbar.setBackgroundColor(Color.TRANSPARENT);
         }
         TypedValue value = new TypedValue();
-        mContext.getTheme().resolveAttribute(android.R.attr.homeAsUpIndicator, value, true);
+        themedContext.getTheme().resolveAttribute(android.R.attr.homeAsUpIndicator, value, true);
         mToolbar.setNavigationIcon(
-                getResources().getDrawable(value.resourceId, mContext.getTheme()));
+                getResources().getDrawable(value.resourceId, themedContext.getTheme()));
 
         mToolbar.getMenu().add(Menu.NONE, MENU_RESET, 0, com.android.internal.R.string.reset)
                 .setShowAsAction(MenuItem.SHOW_AS_ACTION_IF_ROOM);
diff --git a/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileViewImpl.kt b/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileViewImpl.kt
index cce67e9f81f2..7586f64034fa 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileViewImpl.kt
+++ b/packages/SystemUI/src/com/android/systemui/qs/tileimpl/QSTileViewImpl.kt
@@ -122,7 +122,7 @@ constructor(
 
     private val colorActive = Utils.getColorAttrDefaultColor(context, R.attr.shadeActive)
     private val colorInactive = Utils.getColorAttrDefaultColor(context, R.attr.shadeInactive)
-    private val colorUnavailable = Utils.getColorAttrDefaultColor(context, R.attr.shadeDisabled)
+    private val colorUnavailable = Utils.applyAlpha(UNAVAILABLE_ALPHA, colorInactive)
 
     private val overlayColorActive =
         Utils.applyAlpha(
-- 
2.43.0

